curl_cffi/curl.py

```diff
@@ -88,32 +92,34 @@ class Curl:
     Wrapper for `curl_easy_*` functions of libcurl.
     """
 
-    def __init__(self, cacert: str = DEFAULT_CACERT, debug: bool = False, handle = None):
+    def __init__(self, cacert: str = DEFAULT_CACERT, debug: bool = False, handle = None, _ffi = ffi_chrome, _lib = lib_chrome):
         """
         Parameters:
             cacert: CA cert path to use, by default, curl_cffi uses its own bundled cert.
             debug: whether to show curl debug messages.
         """
-        self._curl = lib.curl_easy_init() if not handle else handle
-        self._headers = ffi.NULL
-        self._resolve = ffi.NULL
+        self._ffi = _ffi
+        self._lib = _lib
```

Comment:
As `Curl` is exposed api, could use auto-detect based on `impersonate` parameter (like requests.Session):
```py
class Curl:
    """
    Wrapper for `curl_easy_*` functions of libcurl.
    """

    def __init__(self, cacert: str = DEFAULT_CACERT, debug: bool = False, handle = None, impersonate = None):
        """
        Parameters:
            cacert: CA cert path to use, by default, curl_cffi uses its own bundled cert.
            debug: whether to show curl debug messages.
        """
        self._ffi, self._lib = get_backend_lib(impersonate)  # return chrome by default
        self._curl = self._lib.curl_easy_init() if not handle else handle
        self._headers = self._ffi.NULL
        self._resolve = self._ffi.NULL
```

Reply:
I'm not sure there is an easy way to achieve this auto-detect - in requests.Session, the impersonate parameter could be set in the constructor, or it could be set (and changed) for each individual request. For example, if a user omits the constructor impersonate, then Session would default to the Chrome version, but what if the user tries to use Firefox in the request itself? I think it's could lead to confusion on how to use or not use this class.

The addition of extra classes make it explicitly clear what the underlying implementation is, and keeps the existing API intact. Firefox is then opt-in by customizing the curl instance and the impersonate parameter.